### PR TITLE
fix: use ui alertbar instead of material-ui

### PR DESCRIPTION
**Pull request by jenniferarnesen**
The following changes have been made in this PR:
* switch from material-ui to @dhis2/ui AlertStack/AlertBar.
** One consequence of this is that we lose the formatting (bold) of the dashboard name in the message.
** Slight visual difference - previously the bar rested on the bottom of the viewport, now it is positioned slightly above the bottom
* rename snackbar to alert for reducers, actions and the component
* simplify the alert redux store to only include the message string (the original code was potentially overengineered)

Before, with material-ui
image

Aftr, with @dhis2/ui AlertBar
image

### CODE REVIEW

File: src/actions/selected.js [hunk 1]
```diff
@@ -75,18 +70,12 @@ export const acClearSelectedItemActiveTypes = () => ({
...
+    const alertTimeout = setTimeout(() => {
+        const name = sGetDashboardById(getState(), id)?.displayName
 
+        if (sGetSelectedIsLoading(getState()) && name) {
             dispatch(
-                acReceivedSnackbarMessage({
-                    message: loadingDashboardMsg,
-                    open: true,
-                })
+                acSetAlertMessage(`${i18n.t('Loading dashboard')}: ${name}`)
```
Comment (edoardo) [inline src/actions/selected.js]: The string should probably have a placeholder for the name.
Some languages might need to position the name before the rest of the text, also, the `: ` can cause problems (ie. RTL)

File: src/components/AlertBar/AlertBar.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { connect } from 'react-redux'
+import { AlertBar, AlertStack } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+
+import { sGetAlertMessage } from '../../reducers/alert'
+import { acClearAlertMessage } from '../../actions/alert'
+
+export const Alert = ({ message, onClose }) =>
+    message ? (
+        <AlertStack>
```
Comment (edoardo) [inline src/components/AlertBar/AlertBar.js:11]: Not sure we need `AlertStack` if we only have 1 message at a time...

File: src/components/AlertBar/AlertBar.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            <AlertBar onHidden={onClose} permanent>
```
Comment (edoardo) [inline src/components/AlertBar/AlertBar.js:12]: For DV I used the `variant` prop for different types of alerts, and the `permanent` prop is only set for errors or warnings.
Ie. you would want the loading message to disappear automatically. That is built in so you don't have to handle the timeout in the app.
See: https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/components/Snackbar/Snackbar.js#L16-L37

**Comment by cooper-joe** [comment]
Alert bar usage LGTM 👍

**Review by cooper-joe — APPROVED** [review]

File: src/actions/selected.js (hunk 1 above)
Comment (jenniferarnesen) [reply src/actions/selected.js]: Then we should find another separator. I can't put ":" into the translation string bc it is used as a namespace separator

File: src/components/AlertBar/AlertBar.js (hunk 2 above)
Comment (jenniferarnesen) [reply src/components/AlertBar/AlertBar.js:11]: The AlertStack provides the positioning, so we need it.

File: src/actions/selected.js (hunk 1 above)
Comment (edoardo) [reply src/actions/selected.js]: True. But regardless of the `:` problem, the interpolation with the placeholder still applies.

File: src/components/AlertBar/AlertBar.js (hunk 2 above)
Comment (edoardo) [reply src/components/AlertBar/AlertBar.js:11]: ok, i should probably update DV then.

File: src/components/AlertBar/AlertBar.js (hunk 3 above)
Comment (jenniferarnesen) [reply src/components/AlertBar/AlertBar.js:12]: The way the alertbar currently works is that the message stays until  the app detects that the dashboard is finished loading. So I actually want the message to be permanent in this case. When the loading is complete, then the message in redux is cleared and the Alert renders null.

File: src/actions/selected.js (hunk 1 above)
Comment (jenniferarnesen) [reply src/actions/selected.js]: @edoardo updated to use interpolation

File: src/components/AlertBar/AlertBar.js (hunk 3 above)
Comment (edoardo) [reply src/components/AlertBar/AlertBar.js:12]: ok, sounds good! 👍

File: src/actions/selected.js [hunk 4]
```diff
@@ -75,18 +70,14 @@ export const acClearSelectedItemActiveTypes = () => ({
...
+        const name = sGetDashboardById(getState(), id)?.displayName
 
+        if (sGetSelectedIsLoading(getState()) && name) {
             dispatch(
-                acReceivedSnackbarMessage({
-                    message: loadingDashboardMsg,
-                    open: true,
-                })
+                acSetAlertMessage(
+                    i18n.t(`Loading dashboard – ${name}`, { name })
```
Comment (edoardo) [inline src/actions/selected.js]: 🤔  this doesn't look like is using the correct syntax:
https://www.i18next.com/translation-function/interpolation

I suspect is not doing what we want?
```suggestion
                    i18n.t("Loading dashboard – {{name}}", { name })
```

Comment (jenniferarnesen) [reply src/actions/selected.js]: Oops, you're right 😅

**Review by edoardo — APPROVED** [review]

**Pull request merged**